src/core/src/bootstrap/ConfigurationFactory.py

```diff
@@ -68,6 +75,9 @@ def __init__(self, log_file_path, real_record_path, recorder_enabled, emulator_e
             'yum_test_config':    self.new_test_configuration(Constants.YUM, YumPackageManager),
             'zypper_test_config': self.new_test_configuration(Constants.ZYPPER, ZypperPackageManager)
         }
+        
+        self.vm_context = self.get_vm_environment()
```

Comment:
To standardize terminology and make it unambiguous (since environment and context can mean many things), maybe use **vm_cloud_type** everywhere -- like:

self.vm_cloud_type = self.get_vm_cloud_type()